sphere-models/src/it/java/io/sphere/sdk/products/search/ProductProjectionSearchModelFacetedSearchIntegrationTest.java

```diff
@@ -0,0 +1,440 @@
+                termStats -> assertThat(termStats).containsExactly(
+                        TermStats.of("2001-09-11", 2),
+                        TermStats.of("2002-10-12", 1)));
+    }
+
+    @Test
+    public void facetedSearchOnDateRangedSetAttributes() throws Exception {
+        testResultWithRange(model().allVariants().attribute().ofDate(ATTR_NAME_DATE_SET).facetedSearch().byRange(atLeast(DATE_2002.toString())),
+                ids -> assertThat(ids).containsOnly(product1.getId()),
+                rangeStats -> {
+                    assertThat(rangeStats.getLowerEndpoint()).isEqualTo("0.0");
+                    assertThat(rangeStats.getUpperEndpoint()).isEqualTo(null);
+                    assertThat(rangeStats.getCount()).isEqualTo(2L);
+                    assertThat(rangeStats.getMin()).isEqualTo("1.0001664E12");
+                    assertThat(rangeStats.getMax()).isEqualTo("1.0001664E12");
+                    assertThat(rangeStats.getSum()).isEqualTo("2.0003328E12");
+                    assertThat(rangeStats.getMean()).isEqualTo(1.0001664E12D);
+                });
+    }
+
+    @Test
+    public void facetedSearchOnTimeSetAttributes() throws Exception {
+        testResultWithTerms(model().allVariants().attribute().ofTime(ATTR_NAME_TIME_SET).facetedSearch().by(TIME_23H.toString()),
+                ids -> assertThat(ids).containsOnly(product1.getId()),
+                termStats -> assertThat(termStats).containsExactly(
+                        TermStats.of("22:05:09.203", 2),
+                        TermStats.of("23:06:10.204", 1)));
+    }
+
+    @Test
+    public void facetedSearchOnTimeRangedSetAttributes() throws Exception {
+        testResultWithRange(model().allVariants().attribute().ofTime(ATTR_NAME_TIME_SET).facetedSearch().byRange(atLeast(TIME_23H.toString())),
+                ids -> assertThat(ids).containsOnly(product1.getId()),
+                rangeStats -> {
+                    assertThat(rangeStats.getLowerEndpoint()).isEqualTo("0.0");
+                    assertThat(rangeStats.getUpperEndpoint()).isEqualTo(null);
+                    assertThat(rangeStats.getCount()).isEqualTo(2L);
+                    assertThat(rangeStats.getMin()).isEqualTo("7.9509203E7");
+                    assertThat(rangeStats.getMax()).isEqualTo("7.9509203E7");
+                    assertThat(rangeStats.getSum()).isEqualTo("1.59018406E8");
+                    assertThat(rangeStats.getMean()).isEqualTo(7.9509203E7D);
+                });
+    }
+
+    @Test
+    public void facetedSearchOnDateTimeSetAttributes() throws Exception {
+        testResultWithTerms(model().allVariants().attribute().ofDateTime(ATTR_NAME_DATE_TIME_SET).facetedSearch().by(DATE_TIME_2002_23H.toString()),
+                ids -> assertThat(ids).containsOnly(product1.getId()),
+                termStats -> assertThat(termStats).containsExactly(
+                        TermStats.of("2001-09-11T22:05:09.203+0000", 2),
```

Comment:
That is weird, dates end with Z: http://dev.sphere.io/http-api-types.html#datetime